test/offline/test_cql.py

```diff
@@ -78,16 +81,26 @@ def test_cql(args=get_args()):
     else:
         buffer = gather_data()
     env = gym.make(args.task)
-    args.state_shape = env.observation_space.shape or env.observation_space.n
-    args.action_shape = env.action_space.shape or env.action_space.n
-    args.max_action = env.action_space.high[0]  # float
+    action_space = cast(gym.spaces.Box, env.action_space)
+
+    action_space_info = get_space_info(action_space)
+    observation_space_info = get_space_info(env.observation_space)
+
+    args.state_shape = observation_space_info.state_shape
```

Comment:
There's some misconception here, the object extracted with `get_space_info(action_space)` should not have a `state_shape` field. 

A better interface might be given from the proposed `from_env` constructor method of SpaceInfo, which can collect both the observation and the action space info. Then you also need only a single SpaceInfo object here to fill the args

Reply:
Fixed. Now caller calls once to get a space_info with all the attrs from action and observation space it needs.